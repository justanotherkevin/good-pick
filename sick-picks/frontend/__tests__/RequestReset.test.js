import { mount } from 'enzyme';
import wait from 'waait';
import toJSON from 'enzyme-to-json';
import { MockedProvider } from 'react-apollo/test-utils';
import RequestReset, {
  REQUEST_RESET_MUTATION
} from '../components/RequestReset';

const mocks = [
  {
    request: {
      query: REQUEST_RESET_MUTATION,
      variables: { email: 'test@test.test' }
    },
    result: {
      data: {
        requestReset: { message: 'success', __typename: 'message' }
      }
    }
  }
];

describe('<RequestReset/> component test', () => {
  it('renders and matches snapshot', () => {
    const wrapper = mount(
      <MockedProvider>
        <RequestReset />
      </MockedProvider>
    );
    const form = wrapper.find('form');
    expect(toJSON(form)).toMatchSnapshot();
    // console.log(form.debug());
  });

  it('calls the mutations', async () => {
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <RequestReset />
      </MockedProvider>
    );
    // simulate input change
    wrapper.find('input').simulate('change', {
      target: {
        name: 'email',
        value: 'test@test.test'
      }
    });
    // simulate form submit
    wrapper.find('form').simulate('submit');
    await wait(5);
    wrapper.update();
    const targetMessage = wrapper.find('p');
    expect(targetMessage.text()).toBe(
      'Success! Check your email for a reset link!'
    );
  });
});
