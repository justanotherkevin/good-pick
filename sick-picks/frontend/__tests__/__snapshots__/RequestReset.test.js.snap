// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestReset/> component test renders and matches snapshot 1`] = `
<form
  className="Form-sc-1xszr8q-0 krRxky"
  method="post"
  onSubmit={[Function]}
>
  <fieldset
    aria-busy={false}
    disabled={false}
  >
    <h2>
      Request a password reset
    </h2>
    <DisplayError
      error={Object {}}
    />
    <label
      htmlFor="email"
    >
      Email
      <input
        name="email"
        onChange={[Function]}
        placeholder="email"
        type="email"
        value=""
      />
    </label>
    <button
      type="submit"
    >
      Request Reset!
    </button>
  </fieldset>
</form>
`;
